trin-core/src/portalnet/types/content_keys.rs

```diff
@@ -216,25 +218,25 @@ impl HistoryContentKey {
     /// Returns the content-type of a HistoryContentKey
     pub fn content_type(&self) -> u8 {
         match self {
-            HistoryContentKey::HeaderKey(_) => 0x01,
-            HistoryContentKey::BodyKey(_) => 0x02,
-            HistoryContentKey::ReceiptsKey(_) => 0x03,
+            HistoryContentKey::HeaderKey(_) => 0x00,
+            HistoryContentKey::BodyKey(_) => 0x01,
+            HistoryContentKey::ReceiptsKey(_) => 0x02,
```

Comment:
Hmmm, a bit concerning that we didn't catch this earlier, but glad we're getting it right now. Just for my understanding, the reason we weren't catching it earlier seems to be because we were just checking the round trip encoding / decoding, and not checking that the intermediate value was compliant with the test vectors?

Reply:
The spec was changed since last time in order to be able to use a union more easily, starting from the 0th selector (instead of the 1st). But yeah, previously I was only testing if the encoding and then decoding returned the original value, was not doing any intermediate checking.